partcad/src/partcad/part_factory_scad.py

```
@@ -13,16 +13,83 @@
+
+
+def _write_render_script(source_path, call_line):
+    """Copy the script to a throwaway file with 'call_line' appended.
+
+    The source is never modified: a library-style SCAD file that only defines a
+    module renders nothing on its own, so a call to that module is appended to a
+    temporary copy that is rendered and then deleted.
+    """
+    with open(source_path, "r") as f:
+        source = f.read()
+    tmp_path = tempfile.mktemp(".scad")
```

Ruff (0.15.21)

[error] Use of insecure and deprecated function (`mktemp`)

(S306)